iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -1175,6 +1177,11 @@ internal string PopulateMessagePropertiesFromMessage(string topicName, Message m
                 msg = topicName;
             }
 
+            if (Encoding.ASCII.GetByteCount(msg) > MaxTopicNameLength)
+            {
+                throw new MessageTooLargeException($"TopicName for MQTT packet cannot be larger than {MaxTopicNameLength} bytes, current size is {Encoding.ASCII.GetByteCount(msg)}. Please send a smaller message or use AMQP. TopicName is \n\"{topicName}\"");
```

Comment:
The message isn't clear: devs don't set topic names. Please explain what are the likely causes for the topic name to be large due to the application using our API.
1. Smaller message is wrong here: sending a smaller payload won't affect topic name.
2. Also, definitely do _not_ leak the topic name itself: it could contain PII/HBI information and the error is caused by input. Doing so could allow someone with a device to display information such as device name, etc.
3. We should display the current len and expected len. Also, using AMQP or HTTP (not only AMQP).

/cc @nayato